tests/periph_i2c/main.c

```diff
@@ -29,253 +29,264 @@
 
 static int i2c_dev = -1;
 
-static i2c_speed_t map_speed[] = {
-    I2C_SPEED_LOW,
-    I2C_SPEED_NORMAL,
-    I2C_SPEED_FAST,
-    I2C_SPEED_FAST_PLUS,
-    I2C_SPEED_HIGH
-};
-
-int cmd_init_master(int argc, char **argv)
+int cmd_i2c_init(int argc, char **argv)
 {
-    int dev, speed, res;
+    int dev, res;
 
-    if (argc != 3) {
+    if (argc != 2) {
         puts("Error: Init: Invalid number of arguments!");
-        printf("Usage:\n%s: [DEVICE] [SPEED]\n", argv[0]);
+        printf("Usage:\n%s: [DEVICE]\n", argv[0]);
         puts("    with DEVICE:");
         for (unsigned i = 0; i < I2C_NUMOF; i++) {
             printf("          %u -> I2C_%u\n", i, i);
         }
-        puts("         SPEED:");
-        puts("          0 -> SPEED_LOW (10kbit/s)");
-        puts("          1 -> SPEED_NORMAL (100kbit/s)");
-        puts("          2 -> SPEED_FAST (400kbit/s)");
-        puts("          3 -> SPEED_FAST_PLUS (1Mbit/s)");
-        puts("          4 -> SPEED_HIGH (3.4Mbit/s)\n");
         return 1;
     }
 
     dev = atoi(argv[1]);
-    speed = atoi(argv[2]);
 
-    if (speed < 0 || speed > 4) {
-        puts("Error: Speed value not in range (0 - 4)");
+    res = i2c_init(dev);
```

Comment:
i2c_init must not be call there. It only belongs to drivers/periph_common/init.c